### ISSUE TITLE
Make the go version check better support multi-version go installs

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/osmosis#3762]
Cref: https://github.com/osmosis-labs/osmosis/pull/3634

This PR breaks all users who have an up-to-date version of `go` on their build hosts. One can [fetch older Go versions](https://go.dev/doc/manage-install) and call them as e.g. `go1.18 build ...`, which would in theory solve the problem here, but it doesn't work because the Makefile doesn't parameterize [calls to the `go` tool](https://github.com/osmosis-labs/osmosis/blob/main/Makefile#L13-L14).

Quick fix would be 

```Makefile
GO ?= go

GO_MAJOR_VERSION = $(shell ${GO} version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f1)
GO_MINOR_VERSION = $(shell ${GO} version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f2)

$(BUILD_TARGETS): check_version go.sum $(BUILDDIR)/
	${GO} $@ -mod=readonly $(BUILD_FLAGS) $(BUILD_ARGS) ./...
```

with equivalent changes (`go` → `${GO}`) throughout the file.

_Originally posted by @08d2 in https://github.com/osmosis-labs/osmosis/issues/3634#issuecomment-1339827480_

**Comment by ValarDragon** [comment]
An alternative is in the build check, to try `go1.18` in the install when go is the wrong major version (and otherwise report an error, and tell the user to run `go1.18`

**Comment by 08d2** [comment]
To be clear, these are temporary hacks, and cannot be relied upon. Osmosis _must_ build and run successfully with the latest major version of Go. Whatever prevents that is a bug that must be fixed.

**Comment by ValarDragon** [comment]
I agree for the sort & compression issues. (The sort should tbh be pretty basic to fix)

If theres FFI issues similar to whats changed in the past, that may genuinely be unavoidable.
